TWLight/resources/views.py

```diff
@@ -61,7 +102,93 @@ def get_context_data(self, **kwargs):
                 _("This partner is not available. You can see it because you "
                     "are a staff member, but it is not visible to non-staff "
                     "users."))
-
+        
```

Comment:
Same as above and because this logic isn't factored out, you've had to write almost the same code twice.